src/main/include/log4cxx/helpers/threadutility.h

```diff
@@ -69,15 +69,14 @@ LOG4CXX_PTR_DEF(ThreadUtility);
 class LOG4CXX_EXPORT ThreadUtility
 {
 	private:
-		ThreadUtility();
-
 		log4cxx::helpers::ThreadStartPre preStartFunction();
 		log4cxx::helpers::ThreadStarted threadStartedFunction();
 		log4cxx::helpers::ThreadStartPost postStartFunction();
 
 		LOG4CXX_DECLARE_PRIVATE_MEMBER_PTR(priv_data, m_priv)
 
 	public:
+		ThreadUtility();
```

Comment:
It seems like this would break the singleton aspect of this class, do we really want it to have a public constructor?

Reply:
Keeping that as a `shared_ptr` makes it similar to other methods(for consistency), but since it is a singleton making it a `ThreadUtility*` would be fine in my opinion as well, since it is guaranteed to never be null in normal operation.